lib/core_ext/numpy.rb

```diff
@@ -59,6 +73,44 @@ def roll(shift, axis: 0)
     end
   end
 
+  def assign(x1, y1, x2, y2, value)
+
+    x1.upto(x2) { |x| y1.upto(y2) { |y| self[y][x] = value } }
+
+    self
+  end
+
+  def paste(x, y, other)
+    raise ArgumentError, 'must be in bounds' if x < 0 || y < 0 || x + other.first.length >= first.length || y + other.length >= length
```

Comment:
if this is a common operation, maybe we should add `valid?` to check a point. there is also `rows`, `cols`, `row_range` and `col_range` from the grid classes. I wonder if we need those in Array too to simplify this stuff. What do you think? Maybe we should avoid polluting Array too much.